dfm-sideline-sidekick-app/download/downloadFromAPI.ts

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import * as FileSystem from "expo-file-system";
+
+import { checkDirectoryExists, checkFileExists } from "./File Preprocess/existenceChecker";
+import { checkDevice } from "./checkDevice/checkDevice";
+import { createResumable } from "./createResumable/createResumable";
+import { getCurrentVersion } from "./versionControl/getCurrentVersion";
+import { getStoredVersion, setStoredVersion } from "./versionControl/storedVersion";
+
+export const downloadJSON = async (fileName: string, OS: string) => {
+  let localhost;
+
+  // check if currently running on an emulator or device
+  const isDevice = checkDevice();
+
+  if (isDevice) {
+    localhost = "100.115.33.192"; // ip address of laptop (for running on physical devices) [1. go to command line, 2. type ipconfig /all 3. it's under IPv4 address]
+  } else {
+    // compatibility for type of device
+    localhost = OS === "android" ? "10.0.2.2" : "127.0.0.1";
+  }
+
+  const url = `http://${localhost}:3001/api/allWithVersion`; // all data
+  const versionUrl = `http://${localhost}:3001/api/version`; // newest version
+
+  // directory in local storage to store files at
+  const fileDir = FileSystem.documentDirectory + "expo/";
+
+  // download resumable object to call functions on
+  const downloadResumable = createResumable(url, fileDir, fileName);
+
+  // checking if file exists, will be null if not
+  const fileExists = await checkFileExists(fileDir, fileName);
+
+  try {
+    // will store the path to our file
+    let uri = "";
+
+    const newestVersion = (await getCurrentVersion(versionUrl))[0].version as string;
+    const storedVersion = await getStoredVersion();
+
+    console.log(); // feel free to remove these extra logs, they're just for output clarity when debugging
+
+    console.log("PRINT VERSIONS BEFORE UPDATE");
+    console.log("UPDATED DEVICE VERSION:", storedVersion);
+    console.log("NEWEST VERSION:", newestVersion);
+
+    console.log();
+
+    // no stored version or current doesn't match stored version
+    if (!fileExists || !storedVersion || storedVersion < newestVersion) {
```

Comment:
change the < to a != in case we switch up version number conventions ever

Reply:
changed!